bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -618,6 +618,17 @@ int handleNavigationStarting(long pView, long pArgs, boolean top) {
 	int hr = args.get_Uri(ppszUrl);
 	if (hr != COM.S_OK) return hr;
 	String url = wstrToString(ppszUrl[0], true);
+	// Feature in WebView2: In case of Browser.setText(), the URL passed into
+	// NavigationStarting / NavigationCompleted is of the form
+	//   data:text/html;<base64_encode_of_the_whole_content>
+	// but ICoreWebView2.get_Source(long[]) returns
+	//   about:blank
+	// Other Browser implementations (IE, WebKit, Safari)
+	// always use 'about:blank' in the LocationEvents sent to LocationListeners
+	// Therefore, to maximize compatibility we also send 'about:blank' here:
+	if (url.startsWith("data:text/html;")) {
+		url = "about:blank";
+	}
```

Comment:
We should check whether this conflicts with #1395

Reply:
Yes, it probably does, with #1395 we won't reach this piece of code anymore.

I have added a comment in #1395.

Reply:
That part is now obsolete, as no more data:text/html URLs occur with #1395 merged.